toolchain/check/testdata/as/adapter_conversion.carbon

```diff
@@ -82,6 +82,33 @@ let b_value: B = ({.x = 1, .y = 2} as A) as B;
 // CHECK:STDERR:
 var b_init: B = ({.x = 1, .y = 2} as A) as B;
 
+// --- fail_init_tuple.carbon
+
+library "[[@TEST_NAME]]";
+
+class Noncopyable {
+  // TODO: Ensure this remains non-copyable once we have rules for class copyability.
+}
+
+class A {
+  adapt (i32, Noncopyable);
+}
+
+fn F(a: A) {
+  let a_value: A = (a as (i32, Noncopyable)) as A;
```

Comment:
We usually put tests that are expected to pass in different file splits from ones that are expected to fail. That way we get extra assurance that we don't have regressions after autoupdate -- if an expected-to-pass test starts to fail, the test will fail even after an autoupdate run.

Reply:
Ok, thanks. I will split off the `fail_init_class.carbon` above as well.

FWIW I tried making a non-`fail_` test in this file fail, ran autoupdate, and then `bazel test` and it still fails that expected-to-pass test regardless of the presence of a `fail_` test in the same file or not. Not sure if that's unexpected then.